interpreter/cling/tools/plugins/clad/CMakeLists.txt

```diff
@@ -12,6 +12,7 @@ set(CLADDIFFERENTIATOR_LIB
   ${_clad_byproduct_binary_dir}/lib/Differentiator/${CMAKE_STATIC_LIBRARY_PREFIX}cladDifferentiator${CMAKE_STATIC_LIBRARY_SUFFIX}
   ${_clad_byproduct_binary_dir}/tools/${CMAKE_STATIC_LIBRARY_PREFIX}cladPlugin${CMAKE_STATIC_LIBRARY_SUFFIX}
   )
+list(JOIN CLANG_INCLUDE_DIRS "|" _clad_clang_include_dirs)
```

Comment:
I am not sure what we are solving here. What is `_clad_clang_include_dirs`? If we are allowing CLANG_INCLUDE_DIRS to be different from `_clad_clang_include_dirs` that'd be incorrect.

Reply:
Disclaimer: I'm not very knowledgeable about cmake so I might have misunderstood how this patch actually works.

The issue is that `DCLANG_INCLUDE_DIRS` can contain multiple items such as `/my/dir1` `/my/dir2` so the `CMAKE_ARGS` argument is expanded to:

```cmake
   CMAKE_ARGS -G ${CMAKE_GENERATOR} -DCLAD_BUILD_STATIC_ONLY=ON
               -DCMAKE_INSTALL_PREFIX=${CLING_PLUGIN_INSTALL_PREFIX}
               -DCLAD_PATH_TO_LLVM_BUILD=${LLVM_BINARY_DIR}
               -DCLANG_INCLUDE_DIRS=/my/dir1 /my/dir2
```

And `/my/dir2` isn't passed through as an include directory when configuring `clad`.

This solution works by setting `_clad_clang_include_dirs` to `/my/dir1|/my/dir2` and then using `LIST_SEPARATOR |` to tell `ExternalProject_Add` to split up the lists.

Reply:
Don't you need to just wrap it into quotes? Eg. `-DCLANG_INCLUDE_DIRS="${CLANG_INCLUDE_DIRS}`

Reply:
Ok, could you use `string(REPLACE...`, instead of JOIN?

Reply:
`list(JOIN ...)` is only available with CMake 3.12 and above, we cannot use it, as we need to support CMake 3.4.3 and above in ROOT.

Reply:
Changed 👍